src/main/java/bartenderbob/Storage.java

```diff
@@ -0,0 +1,180 @@
+                //Create tasks.txt
+                Files.createFile(path);
+            }
+            return taskArray;
+        } catch (IOException e) {
+            throw new BartenderBobException();
+        }
+    }
+
+    private Task parseTaskFromString(String taskString) {
+        String[] taskStringComponents = taskString.split(" \\| ");
+        String taskType = taskStringComponents[0];
+        String taskStatus = taskStringComponents[1];
+        boolean isDone = taskStatus.equals("X");
+        String description = taskStringComponents[2];
+
+        if (taskType.equals("T")) {
+            return new ToDo(description, isDone);
+        } else if (taskType.equals("D")) {
+            String dueDate = taskStringComponents[3];
+            dueDate = convertDateFormat(dueDate);
+            return new Deadline(description, dueDate, isDone);
+        } else if (taskType.equals("E")) {
+            String fromDate = taskStringComponents[3];
+            fromDate = convertDateFormat(fromDate);
+            String toDate = taskStringComponents[4];
+            toDate = convertDateFormat(toDate);
+            return new Event(description, fromDate, toDate, isDone);
+        }
+        return null;
+    }
+
+    /**
+     * Save changes to the files in the hard disk
+     * especially after unmarking, marking and deleting them.
+     *
+     * @param tasks The tasks that the user has inputted to BartenderBob.
+     */
+    public void saveChanges(ArrayList<Task> tasks) {
+        try {
+            assert filePath != null : "File path cannot be null";
+            assert tasks != null : "Task cannot be null";
+            Path path = Paths.get(filePath);
+            Files.write(path, "".getBytes(), StandardOpenOption.TRUNCATE_EXISTING);
+            for (Task task : tasks) {
+                saveTask(task);
+            }
+        } catch (IOException e) {
+            ui.showSaveChangesError();
+        }
+
+    }
+
+    /**
+     * Convert date format from MMM dd yyyy to yyyy-MM-dd
+     *
+     * @param oldDateFormat Date which is of MMM dd yyyy format.
+     * @return Date which is of yyyy-MM-dd format.
+     */
+    private String convertDateFormat(String oldDateFormat) {
+        //The storage and show() method stores the date as "MMM dd yyyy"
+        //however, the constructor for deadline and event takes in a date
+        //as "yyyy-MM-dd". That's why we need this method when loading from
+        //storage into the chatbot.
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+
+        // Parse the original string to LocalDate
+        LocalDate localDate = LocalDate.parse(oldDateFormat, inputFormatter);
+
+        // Define the formatter for the output pattern
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        // Format the LocalDate to the desired output pattern
+        return localDate.format(outputFormatter);
```

Comment:
I like that you make multiple comments for confusing methods, making the code easy to read.